### PR TITLE
BigNumber, intcomma fixes from SpankChain

### PR DESCRIPTION
**Pull request by wolever**

### CODE REVIEW

File: modules/client/src/controllers/BuyController.ts [hunk 1]
```diff
@@ -69,6 +69,7 @@ export default class BuyController extends AbstractController {
           // PUNT on this -- AB
           break
         case 'PT_CHANNEL':
+        case 'PT_CUSTODIAL':
```
Comment (LayneHaber) [inline modules/client/src/controllers/BuyController.ts]: So my guess is this won't cause any material problems, but it should be noted this type isn't implemented hub-side with this PR

File: modules/client/src/register/bignumber.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import {BigNumber} from "bignumber.js"
+
+const oldBigNumberToString = BigNumber.prototype.toString
+BigNumber.prototype.toString = function(base?: number) {
+  // BigNumber.toString will only use exponential notation if a base is not
+  // specified. In dev, throw an error so it can be replaced with `.toFixed()`,
+  // or in production, log an error and force base 10.
+  if (!base) {
+    const err = new Error('Potentially unsafe of BigNumber.toString! Use .toFixed() instead.')
```
Comment (LayneHaber) [inline modules/client/src/register/bignumber.ts]: 🥇

File: modules/client/src/types.ts [hunk 3]
```diff
@@ -945,7 +945,7 @@ Returns:
 
 */
 
-export type PurchasePaymentType = 'PT_CHANNEL' | 'PT_THREAD' | 'PT_LINK'
+export type PurchasePaymentType = 'PT_CHANNEL' | 'PT_THREAD' | 'PT_CUSTODIAL' | 'PT_LINK'
```
Comment (LayneHaber) [inline modules/client/src/types.ts]: same note as above re: hub types

File: modules/client/src/controllers/BuyController.ts (hunk 1 above)
Comment (rhlsthrm) [reply modules/client/src/controllers/BuyController.ts]: Yep, should just fail on the hub immediately.

**Pull request merged**